src/Documentation/Getting-Started-With-Orleans/Grain-LifeCycle.md

```diff
@@ -0,0 +1,65 @@
+---
+layout: page
+title: Grain LifeCycle
+---
+
+
+## Grain LifeCycle
+
+We say grains virtually always exist but here we want to describe different stages of a grain's life.
```

Comment:
This is confusing verbiage to me. I'm afraid that the analogies, such as "like a class definition" and "instantiated with a `new` keyword in C#" will likely only contribute to the confusion. I think we can be much more straightforward here. For example, something along the following lines.

"Grains are logical entities that always exist, virtually, and have stable logical identities (keys). Application code never creates or destroys grains. Instead, it acts as if all possible grains are always in memory and available for processing requests.

Grains get physically instantiated, activated, by the Orleans runtime automatically on an as-needed to process incoming requests. After a grain has been idle for a certain amount of time, the Orleans runtime automatically removes, deactivates, it from memory.

A physical instances of a grain in memory is called a grain activation. Grain activations are invisible to application code as well as the process of activating and deactivating them. Only the grain itself can be aware of that  - by overriding virtual methods `OnActivateAsync` and `OnDeactivateAsync` that get invoked upon activation and deactivation of the grain respectively.

Over the course of its eternal, virtual, life a grain goes through the cycles of activations and deactivations, staying always available for callers to invoke it, whether it is in memory at the time of the call or not."